pkg/plugins/runtime/k8s/controllers/gateway_instance_controller.go

```diff
@@ -73,11 +75,14 @@ func (r *GatewayInstanceReconciler) Reconcile(ctx context.Context, req kube_ctrl
 
 	updateStatus(gatewayInstance, svc, deployment)
 
-	if err := r.Client.Status().Patch(ctx, gatewayInstance, kube_client.MergeFrom(orig)); err != nil {
```

Comment:
I think the right way to handle this is using `SetStatusCondition` [like in the gatewayapi controllers](https://github.com/kumahq/kuma/blob/57884e62ff442968f04c003f94bad9cbcd65076c/pkg/plugins/runtime/k8s/controllers/gatewayapi/gateway_status.go#L231)